com.ibm.streamsx.topology/opt/python/packages/streamsx/scripts/streamtool.py

```diff
@@ -26,17 +26,29 @@ def _submitjob_parser(subparsers):
     job_submit.add_argument('sabfile', help='Location of sab file.', metavar='sab-pathname')
     job_submit.add_argument('--jobConfig', '-g', help='Specifies the name of an external file that defines a job configuration overlay')
     job_submit.add_argument('--jobname', help='Specifies the name of the job.')
-    job_submit.add_argument('--P', '-P', help='Specifies a submission-time parameter and value for the job')
+    job_submit.add_argument('--P', '-P', help='Specifies a submission-time parameter and value for the job', action='append')
     _user_arg(job_submit)
 
 def _submitjob(instance, cmd_args):
     """Submit a job."""
     job_config = None
-    if cmd_args.jobname:
-        job_config = streamsx.topology.context.JobConfig(job_name=cmd_args.jobname)
+
     if cmd_args.jobConfig:
         with open(cmd_args.jobConfig) as fd:
             job_config = streamsx.topology.context.JobConfig.from_overlays(json.load(fd))
+    else:
+        job_config = streamsx.topology.context.JobConfig()
+
+    if cmd_args.jobname:
+        job_config.job_name = cmd_args.jobname
+
+    if cmd_args.P:
+        for param in cmd_args.P:
+            name_value_pair = param.split("=")
+            if (len(name_value_pair) != 2):
+                raise Exception("The format of the following submission-time parameter is not valid: {}. The correct syntax is: <name>=<value>".format(param))
```

Comment:
Should be `ValueError`

https://docs.python.org/3/library/exceptions.html#ValueError